src/Model/Resolver/Slider.php

```diff
@@ -88,48 +69,62 @@ public function resolve(
-
-            $maps = $this->mapCollectionFactory->create();
-            $maps = $maps->addSliderFilter($args['id'])
-                ->addIsActiveFilter()
-                ->getItems();
-
-            foreach ($sliderData['slides'] as &$slide) {
-                if (array_key_exists('mobile_image', $slide) && isset($slide['mobile_image'])){
-                    $slide['mobile_image'] = DirectoryList::MEDIA . DIRECTORY_SEPARATOR . $slide['mobile_image'];
-                }
-                if (array_key_exists('desktop_image', $slide) && isset($slide['desktop_image'])){
-                    $slide['desktop_image'] = DirectoryList::MEDIA . DIRECTORY_SEPARATOR . $slide['desktop_image'];
-                }
-                foreach ($maps as $map) {
-                    if ($map['slide_id'] === $slide['slide_id']) {
-                        $slide['maps'][] = $map;
-                    }
-                }
+        if (!isset($args['id'])) {
+            return $this->valueFactory->create($result);
+        }
+
+        $id = $args['id'];
+        try {
+            /** @var \Scandiweb\Slider\Model\Slider $slider */
+            $slider = $this->sliderRepository->get($id);
+        } catch (NoSuchEntityException $e) {
+            $result = function () use ($id) {
+                return new GraphQlNoSuchEntityException(__(`Slider with id "{$id}" does not exist.`));
+            };
+
+            return $this->valueFactory->create($result);
+        }
+
+        $sliderData = $slider->getData();
+        $sliderData['slides'] = [];
+        $slides = $slider->getSlides();
+
+        /** @var \Scandiweb\Slider\Model\Slide $slide */
+        foreach ($slides as $slide) {
+            $slideData = $slide->getData();
+            $slideData['maps'] = $slide->getMaps();
+
+            if ($slide->getFirstMobileImageLocation()) {
+                $slideData[$slide::FIRST_MOBILE_IMAGE] = $slide->getFirstMobileImageUrl();
+            }
+
+            if ($slide->getFirstDesktopImageLocation()) {
+                $slideData[$slide::FIRST_DESKTOP_IMAGE] = $slide->getFirstDesktopImageUrl();
+            }
+
+            if ($slide->getSecondMobileImageLocation()) {
+                $slideData[$slide::SECOND_MOBILE_IMAGE] = $slide->getSecondMobileImageUrl();
+            }
+
+            if ($slide->getSecondDesktopImageLocation()) {
+                $slideData[$slide::SECOND_DESKTOP_IMAGE] = $slide->getSecondDesktopImageUrl();
+            }
+
+            if ($slide->getThirdMobileImageLocation()) {
+                $slideData[$slide::THIRD_MOBILE_IMAGE] = $slide->getThirdMobileImageUrl();
             }
-            unset ($slide);
 
-            if ($sliderData) {
-                $result = function () use ($sliderData) {
-                    return $sliderData;
-                };
+            if ($slide->getThirdDesktopImageLocation()) {
+                $slideData[$slide::THIRD_DESKTOP_IMAGE] = $slide->getThirdDesktopImageUrl();
             }
```

Comment:
Why is this needed?
I guess if there is no image we should get null in response and can handle that on FE.
Isn't it?

Reply:
I don't know how it works on FE, so I didn't change any logic here for compatibility.